### ISSUE TITLE
Using $this->whenLoaded causes 502 Bad Gateway

### ISSUE THREAD

**Issue by antonhedling** [dedoc/scramble#31]
No idea why this is happening, but using `$this->whenLoaded()` in a HTTP Resources causes the /docs/api.json route to throw a 502 Bad Gateway error. Example Resource:

```
public function toArray($request)
{
    return [
        'id' => $this->id,
        'name' => $this->name,
        'subareas' => $this->subareas,
        'schools' => SchoolResource::collection($this->whenLoaded('schools')),
    ];
}
 ```
 
 I've tried debugging but can't really seem to grasp it.

**Comment by romalytvynenko** [comment]
Hey @antonhedling!

Thanks for reporting!

If you found an endpoint causing this, can you please send me more code?

1. Full resource class' source (both `SchoolResource` and the resource which contains `schools`)
2. Full controller class (or, at least, a method code)

This should be a good point for me to start.

Thanks!

**Comment by antonhedling** [comment]
Okay, I did just now find out the line that is causing this, so it looks like this isn't necessarily connected to `$this->whenLoaded()`.

It's when two resources cross-reference one another:

AreaResource.php
```
<?php

namespace App\Http\Resources;

use Illuminate\Http\Resources\Json\JsonResource;

class AreaResource extends JsonResource
{
    /**
     * Transform the resource into an array.
     *
     * @param  \Illuminate\Http\Request  $request
     * @return array|\Illuminate\Contracts\Support\Arrayable|\JsonSerializable
     */
    public function toArray($request)
    {
        return [
            'id' => $this->id,
            'name' => $this->name,
            'subareas' => $this->subareas,
            'schools' => SchoolResource::collection($this->whenLoaded('schools')),
        ];
    }
}
```

SchoolResource.php
```
<?php

namespace App\Http\Resources;

use Illuminate\Http\Resources\Json\JsonResource;

class SchoolResource extends JsonResource
{
    /**
     * Transform the resource into an array.
     *
     * @param  \Illuminate\Http\Request  $request
     * @return array|\Illuminate\Contracts\Support\Arrayable|\JsonSerializable
     */
    public function toArray($request)
    {
        return [
            'id' => $this->id,
            'name' => $this->name,
            'area' => new AreaResource($this->area),
        ];
    }
}
```

If I remove the AreaResource there, it works again. 

I mean sure, you can argue that resources shouldn't look like this. But it would be great if we could prevent whatever infinite loop is caused here.

**Comment by romalytvynenko** [comment]
Awesome, thank you so much! Fix is on the way. This is exactly the problem.

Will ping you here once done.

**Comment by romalytvynenko** [comment]
Hey @antonhedling 

Just released `0.3.0`, please update: https://github.com/dedoc/scramble/releases/tag/v0.3.0

The issue should be fixed.

Thanks for your time and reporting the issue.
